python_modules/dagster/dagster/_core/snap/execution_plan_snapshot.py

```diff
@@ -93,6 +94,19 @@ def __new__(
             ),
         )
 
+    @property
+    def asset_selection(self) -> AbstractSet[AssetKey]:
+        asset_keys = set()
+
+        for step in self.steps:
+            if step.key in self.step_keys_to_execute:
+                for output in step.outputs:
+                    asset_key = check.not_none(output.properties).asset_key
+                    if asset_key:
+                        asset_keys.add(asset_key)
```

Comment:
i think you mentioned there was a similar block of code this was modeled after? should that be replaced by calling this?

Reply:
there is, but it also introspects properties of the outputs while emitting the events: https://github.com/dagster-io/dagster/blob/b88560a0956c8f2d48e3242a02f33135569637ec/python_modules/dagster/dagster/_core/instance/__init__.py#L1430-L1434